### DISCUSSION TITLE
How to create Instance of a Service without Dependency Injection in storyBook

### DISCUSSION THREAD

**Discussion by gurramswadeep** [storybookjs/storybook#17461] [Help]
I have spinner service that toggles spinner based on true/false values.

I want to use a method based decorator to handle this toggling of Spinner Service. So, to get an instance I have created instance of the spinner Service in itself and provided as 'root' as below:

spinner-service.service.ts
`  static instance: SpinnerService; 

  constructor() {
    if (!SpinnerService._singleton) {
      SpinnerService.instance = this;
    }
  }`

 spinner.decorator.ts
 `return (targetClass, fnName, descriptor) => {
    const originalMethod = descriptor.value;

    descriptor.value = function (...args): Observable<any> {
      SpinnerService.instance.active = true;
      return originalMethod.apply(this, args).pipe(finalize(() => (SpinnerService.instance.active = false)));
    };
    return descriptor;
  };`

In decorator, I am accessing the instance of SpinnerService as  "SpinnerService.instance". I am able to run this in angular. But When I use the same demo component in storyBook, the instance of  "SpinnerService" is not getting created because of which the spinner is not getting active/toggled. Is this a limitation of storybook or is there any way to fix this?

**Comment by vvs09** [comment]
@gurramswadee, did you configure it out ?
